debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java

```diff
@@ -251,27 +251,30 @@ private void processReplicationMessages(PostgresPartition partition, PostgresOff
             throws SQLException, InterruptedException {
 
         final Lsn lsn = stream.lastReceivedLsn();
-
+        LOGGER.trace("Processing replication message {}", message);
         if (message.isLastEventForLsn()) {
             lastCompletelyProcessedLsn = lsn;
         }
 
         // Tx BEGIN/END event
         if (message.isTransactionalMessage()) {
+
+            offsetContext.updateWalPosition(lsn, lastCompletelyProcessedLsn, message.getCommitTime(), toLong(message.getTransactionId()),
```

Comment:
I'd be really really careful here. Please make sure that the LSN of the commit is different form the LSN of the following message otherwise we might end up with message loss on crash.

Reply:
```sql
SELECT * FROM pg_create_logical_replication_slot('regression_slot', 'test_decoding');

begin 
	
INSERT INTO public.test
(a)
VALUES('test');
commit

	
INSERT INTO public.test
(a)
VALUES('test2');

SELECT * FROM pg_logical_slot_peek_changes('regression_slot', NULL, NULL);
```

|lsn|xid|data|
|---|---|----|
|0/176A578|558|BEGIN 558|
|0/176A578|558|table public.test: INSERT: a[character varying]:'test'|
|0/176A5E8|558|COMMIT 558|
|0/176A620|559|BEGIN 559|
|0/176A620|559|table public.test: INSERT: a[character varying]:'test2'|
|0/176A690|559|COMMIT 559|


Is this answering your question?